addons/resource/models/resource_resource.py

```diff
@@ -166,7 +166,7 @@ def _get_calendars_validity_within_period(self, start, end, default_company=None
             resource_calendars_within_period[resource.id][calendar] = Intervals([(start, end, self.env['resource.calendar.attendance'])])
         return resource_calendars_within_period
 
-    def _get_valid_work_intervals(self, start, end, calendars=None):
+    def _get_valid_work_intervals(self, start, end, calendars=None, compute_leaves=True):
```

Comment:
Can you reword your commit to avoid using "I"/"We" as mentioned in the guidelines.

```txt
[IMP] resource: add ignore leaves option when getting valid work intervals

This commit adds `compute_leaves` optional param in 
`_get_valid_work_intervals` method to be able to get valid work intervals 
without taking leaves into account.

Related PR:
Enterprise: odoo/enterprise#33549

task-3054342
```

Thanks in advance! 🙂